pkg/trace/sampler/signature.go

```diff
@@ -91,3 +89,34 @@ func computeSpanHash(span *pb.Span, env string, withResource bool) spanHash {
 	}
 	return spanHash(h.Sum32())
 }
+
+// sum32a is lifted from golang fnv package but simplified
+// for our use case to remove interfaces which caused unnecessary allocations.
+type sum32a uint32
+
+const (
+	offset32 = 2166136261
+	prime32  = 16777619
+)
+
+func new32a() sum32a {
+	return offset32
+}
+
+func (s *sum32a) Write(data []byte) {
+	hash := *s
+	for _, c := range data {
+		hash ^= sum32a(c)
+		hash *= prime32
+	}
+	*s = hash
+}
+
+func (s *sum32a) WriteChar(c byte) {
```

Comment:
```suggestion
func (s *sum32a) WriteByte(c byte) {
```
This will match other similar structs in Go such as `bytes.Buffer` or `strings.Builder`